evals/completion_fns/cot.py

```diff
@@ -0,0 +1,68 @@
+"""
+Extending OpenAI Models with Chain-of-Thought Inner Monologue
+"""
+from evals.prompt.base import ChatCompletionPrompt
+from evals.record import record_sampling
+from evals.registry import Registry
+
+DEFAULT_COT_TEMPLATE = "\nBefore answering, reason in a step-by-step manner as to get the right answer, then conclude with the answer."
+DEFAULT_EXTRACT_ANSWER_TEMPLATE = (
+    "\nGiven the above reasoning, the answer in the format requested by the question is:"
+)
+
+
+class ChainOfThoughtCompletionResult:
+    def __init__(self, response) -> None:
+        self.response = response
+
+    def get_completions(self) -> list[str]:
+        return [self.response.strip()]
+
+
+class ChainOfThoughtCompletionFn:
+    def __init__(
+        self,
+        cot_template=DEFAULT_COT_TEMPLATE,
+        extract_answer_template=DEFAULT_EXTRACT_ANSWER_TEMPLATE,
+        cot_completion_fn=None,
+        extract_completion_fn=None,
+        registry_path=None,
+        extra_options=None,
+        **kwargs
+    ) -> None:
+
+        self.extra_options = extra_options
+
+        registry = Registry()
+        if registry_path:
+            registry.add_registry_paths(registry_path)
+
+        if extract_completion_fn is None:
+            extract_completion_fn = cot_completion_fn
+
+        # This model will use chain of thought to answer the question
+        self.cot_template = cot_template
+        self.cot_completion_fn_instance = registry.make_completion_fn(cot_completion_fn)
+
+        # This model will extract the answer from the chain of thought
+        self.extract_answer_template = extract_answer_template
+        self.extract_completion_fn_instance = registry.make_completion_fn(extract_completion_fn)
```

Comment:
For simplicity, should we just have one completion_fn variable for both cot and extract? (until there's a need for using different instances)